### ISSUE TITLE
INSERT/UPDATE/REMOVE AQL - Statements should (optionally) return a value

### ISSUE THREAD

**Issue by nvdnkpr** [arangodb/arangodb#963]
As I wrote in Google Groups (https://groups.google.com/forum/#!topic/arangodb/V0vqXlkCrcg)

the `INSERT/UPDATE/REPLACE/REMOVE` statements should return a value.

This value can vary (can be the whole doc or just the `_id/_key/_rev` - set) but should stay the same for all three.

Also it should be optional, depending if it's assigned to another value or not
e.h.
```
LET newDoc = INSERT { "hello":"world"} IN myCollection
```
should return a value (which is then stored in newDoc and can be RETURNed.

whereas

```
INSERT { "hello":"world" } IN myCollection
```
would be a fire & forget (performance!)

Greetings

Navid

Edit: added REPLACE to the list :)

**Comment by jsteemann** [comment]
This feature has been implemented since 2.4.0.
The syntax is slightly different than suggested above, but it should solve the mentioned problems anyway:

A regular AQL `INSERT` query still does not return any documents:
```
INSERT { "hello" : "world" } IN myCollection
```
However, adding a bit a the end will make it return the created document(s):
```
INSERT { "hello" : "world" } IN myCollection LET result = NEW RETURN result
```
This will also work if a query has inserted more than one document, e.g. the following `INSERT` query will return 100 documents
```
FOR i IN 1..100
  INSERT { "value" : i } IN myCollection LET result = NEW RETURN result
```

The same mechanism works for `REMOVE`, `UPDATE` and `REPLACE` queries, with `UPDATE` and `REPLACE` being able to return either the documents before update/replace or after that.

**Comment by nvdnkpr** [comment]
Very cool. Thank you guys for your awesome work
